### PR TITLE
Switch FindLocalCouncil and LocalTransaction controllers to use locations-api

### PR DESCRIPTION
**Pull request by catalinailie**
Frontend controllers (FindLocalCouncil, LocalTransaction and Licence) need to use Locations API in place of Mapit.

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️
Trello card: https://trello.com/c/6TO6GDJM/2934-switch-frontend-to-use-locations-api-5

### CODE REVIEW

File: test/integration/find_local_council_test.rb [hunk 1]
```diff
@@ -176,37 +168,35 @@ class FindLocalCouncilTest < ActionDispatch::IntegrationTest
...
-          visit "/find-local-council"
-          fill_in "postcode", with: "SW1A 1AA"
-          click_on "Find"
-        end
-
-        should "show advisory message that we have no URL" do
-          assert page.has_content? "We don't have a link for their website."
-        end
-      end
+      # context "when finding a local council without homepage" do
```
Comment (ChrisBAshton) [inline test/integration/find_local_council_test.rb:171]: Should this be commented out?

Comment (catalinailie) [reply test/integration/find_local_council_test.rb:171]: Yes, it should. But first `stub_local_links_manager_has_a_local_authority_without_homepage` needs to accept a `local_custodian_code` as parameter. This was done in here https://github.com/alphagov/gds-api-adapters/pull/1156/commits/be0e0ebc041403d92db0d9a18ad127757288aa63

File: app/controllers/local_transaction_controller.rb [hunk 2]
```diff
@@ -109,6 +115,7 @@ def interaction_details
       }
     else
       @_interaction ||= Frontend.local_links_manager_api.local_link(council, lgsl, lgil)
+      # @_interaction ||= Frontend.local_links_manager_api.local_link_by_custodian_code(local_custodian_code, lgsl, lgil)
```
Comment (ChrisBAshton) [inline app/controllers/local_transaction_controller.rb:118]: This can be deleted

File: test/functional/local_transaction_controller_test.rb [hunk 3]
```diff
@@ -162,13 +140,17 @@ class LocalTransactionControllerTest < ActionController::TestCase
...
         setup do
-          stub_mapit_has_a_postcode_and_areas(
+          stub_locations_api_has_location(
             "BT1 3QG",
-            [0, 0],
             [
-              { "name" => "Belfast City Council", "type" => "LGD", "govuk_slug" => "belfast", "country_name" => "Northern Ireland" },
+              {
+                "latitude" => 0,
+                "longitude" => 0,
```
Comment (ChrisBAshton) [inline test/functional/local_transaction_controller_test.rb:148]: I don't think these values are used, are they? 
Can we remove them and have `stub_locations_api_has_location` set defaults for latitude/longitude if not provided?

File: test/integration/licence_test.rb [hunk 4]
```diff
@@ -142,7 +139,7 @@ class LicenceTest < ActionDispatch::IntegrationTest
           ]
 
           stub_licence_exists(
-            "1071-5-1/00BK",
+            "1071-5-1/00AG",
```
Comment (ChrisBAshton) [inline test/integration/licence_test.rb:142]: Can you explain in the commit message why you had to change these?
I'm guessing it's because the tests previously used incorrect values?

File: app/controllers/licence_controller.rb [hunk 5]
```diff
@@ -61,8 +61,17 @@ def licence_details_from_api_for_local_authority
     licence_details_from_api(snac_from_slug)
   end
 
+  def lgsl
+    content_item_hash["details"]["lgsl_code"]
+  end
+
+  def lgil
+    content_item_hash["details"]["lgil_code"] || content_item_hash["details"]["lgil_override"]
```
Comment (ChrisBAshton) [inline app/controllers/licence_controller.rb:69]: This `override` logic surprised me, and looks new - could you explain why these changes were necessary (in the commit message)?

**Review by ChrisBAshton — COMMENTED** [review]
Well done making these fiddly changes! Just a few minor comments below.

File: test/functional/local_transaction_controller_test.rb (hunk 3 above)
Comment (catalinailie) [reply test/functional/local_transaction_controller_test.rb:148]: No, we don't use anywhere these coordinates.

File: app/controllers/licence_controller.rb (hunk 5 above)
Comment (catalinailie) [reply app/controllers/licence_controller.rb]: `lgil_override` is deprecated, so I removed it from here. This line was copied from local_transaction_controller, but there was added years ago.

File: docs/elections-api.md [hunk 6]
```diff
@@ -2,16 +2,12 @@
...
-lookups via Mapit and then get council and URL information from Local Links
+lookups via Locations API and then get council and URL information from Local Links
 Manager.
 
 Electoral Commission have [a public API](https://api.electoralcommission.org.uk/)
 for finding this information which they keep up to date.
 
-The API is also capable of disambiguating postcodes that span local authority
-boundaries which Mapit is not. (There is work underway to fix this for other
- postcode related services on GOV.UK by using a different postcode lookup product).
```
Comment (ChrisBAshton) [inline docs/elections-api.md:13]: Might be worth keeping some of this:

```suggestion
The API is also capable of disambiguating postcodes that span local authority	
boundaries.
```

**Review by ChrisBAshton — APPROVED** [review]
Nice work ⭐ 👏 

One non-blocking nitpick comment below. Feel free to merge & deploy (probably safest to do that tomorrow morning).
